### ISSUE TITLE
Question about hyper parameter in luna16 dataset

### ISSUE THREAD

**Issue by WeebOwO** [zunzhumu/CPMNetv2#11]
First of all, thank you for your willingness to share the code, it is very helpful to me. I am tyring to reproduce the result in luna16 and not hit the 0.956 froc.
So I am very interested in the hyperparameter settings on Luna16. Can you give me the corresponding hyperparameter settings? 
Looking forward to your prompt reply! 😊
Ps: I know you're probably very busy, so I don't need the full Luna training code
@zunzhumu

**Comment by zxy630** [comment]
Hi, have you complete the train code on LUNA16? 
May I refer to it?
Thanks!

**Comment by WeebOwO** [comment]
Sure, what't your email? I can give you a private copy of my code.

**Comment by zxy630** [comment]
That's so kind! My email is 1481015567@qq.com. Thanks!

**Comment by vincentwan0930** [comment]
Hi, I also want to reproduce the result in luna16.
May I refer to your training code?
My email is 18652996481@163.com
Thank you very much!

**Comment by Pang-b0** [comment]
> Sure, what's your email? I can give you a private copy of my code.

Hello, I'm also interested in your training code for luna16. Can you provide me with a copy? I would appreciate it. My email address is 1399308695@qq.com

**Comment by WeebOwO** [comment]
@vincentwan0930 @zxy630 @Pang-b0  Hi, guys, I open my code at https://github.com/WeebOwO/LuCPMV2.  I will try do some document work so that you can easy to get started.
PS: There should be some bugs, and I hope you can help maintain the repo together😊

**Comment by zunzhumu** [comment]
@WeebOwO I noted the luna16 data needed to be resampled  to 1mm x 1mm x 1mm spacing，and redirected some cases to the standard direction. Finally selected the best model using Froc metric instead of  AP or Acc. The 10-fold evaluation results show in the [https://github.com/zunzhumu/CPMNetv2/blob/master/Froc_10folders.jpg](url)

**Comment by Pang-b0** [comment]
> @WeebOwO I noted the luna16 data needed to be resampled to 1mm x 1mm x 1mm spacing，and redirected some cases to the standard direction. Finally selected the best model using Froc metric instead of AP or Acc. The 10-fold evaluation results show in the [https://github.com/zunzhumu/CPMNetv2/blob/master/Froc_10folders.jpg](url)

Hello author, I want to know how to reproduce the results of LUNA16. Can you provide a training test code on LUNA16?

**Comment by zunzhumu** [comment]
@WeebOwO ![image](https://github.com/zunzhumu/CPMNetv2/assets/24983216/ef7fba7f-52fc-4cd5-a476-d5d3f1ebf9e0)

**Comment by WeebOwO** [comment]
I did the same preprocessing operation and I looked at your preprocessing code, and it's basically no different from Deeplung, NoduleNet. So I don't think my data preprocessing is wrong.

**Comment by WeebOwO** [comment]
I trust your result is true, but It is inevitable that there will be deviations in my own reproduction.  So the best way is relase training code at Luna16 or give some hyperparameter settings. 😚

**Comment by zunzhumu** [comment]
> > @WeebOwO I noted the luna16 data needed to be resampled to 1mm x 1mm x 1mm spacing，and redirected some cases to the standard direction. Finally selected the best model using Froc metric instead of AP or Acc. The 10-fold evaluation results show in the [https://github.com/zunzhumu/CPMNetv2/blob/master/Froc_10folders.jpg](url)
> 
> Hello author, I want to know how to reproduce the results of LUNA16. Can you provide a training test code on LUNA16?
Because I reconstructed the CPMNet v2 in the new company these training codes focused on the specific business. The original CPMNet v2 code for LUNA16 had a different style compared to the current code, so I did not pull that code into this repository

**Comment by zunzhumu** [comment]
@Pang-b0 Because I reconstructed the CPMNet v2 in the new company these training codes focused on the specific business. The original CPMNet v2 code for LUNA16 had a different style compared to the current code, so I did not pull that code into this repository

**Comment by zunzhumu** [comment]
> I trust your result is true, but It is inevitable that there will be deviations in my own reproduction. So the best way is relase training code at Luna16 or give some hyperparameter settings. 😚

So what performance for LUNA16 you can achieved ?@WeebOwO

**Comment by WeebOwO** [comment]
In my test, it's 0.895. @zunzhumu

**Comment by WeebOwO** [comment]
![image](https://github.com/zunzhumu/CPMNetv2/assets/50624245/51d9a8f7-dc6e-4044-9143-6cec3ad9e33e)

**Comment by WeebOwO** [comment]
The network and loss is easy to follow, so I just only curious about hyperparameters.

**Comment by zunzhumu** [comment]
> In my test, it's 0.895. @zunzhumu

Could you provide detailed info for subset 9 validation? For example, in which patients, which nodules were not detected? Based on my experience, your preprocessing code may have only redirected the image but did not adjust the coordinates of the bounding box, which was a bug in the previous public code on GitHub.

**Comment by Pang-b0** [comment]
> 因为我在新公司中重建了CPMNet v2，所以这些培训代码侧重于特定的业务。LUNA2 的原始 CPMNet v16 代码与当前代码的样式不同，因此我没有将该代码拉入此存储库

Thank you for your answer. I noticed this code because of the article SCPM-net. What is the relationship between CPMV2 and SCPM? Is there any chance for the code of SCPMnet to be made public?

**Comment by WeebOwO** [comment]
[0.txt](https://github.com/zunzhumu/CPMNetv2/files/12853834/0.txt)
[1.txt](https://github.com/zunzhumu/CPMNetv2/files/12853835/1.txt)
[2.txt](https://github.com/zunzhumu/CPMNetv2/files/12853836/2.txt)
[3.txt](https://github.com/zunzhumu/CPMNetv2/files/12853837/3.txt)
[4.txt](https://github.com/zunzhumu/CPMNetv2/files/12853838/4.txt)
[5.txt](https://github.com/zunzhumu/CPMNetv2/files/12853839/5.txt)
[6.txt](https://github.com/zunzhumu/CPMNetv2/files/12853840/6.txt)
[7.txt](https://github.com/zunzhumu/CPMNetv2/files/12853841/7.txt)
[8.txt](https://github.com/zunzhumu/CPMNetv2/files/12853842/8.txt)
[9.txt](https://github.com/zunzhumu/CPMNetv2/files/12853843/9.txt)
All my results are made in the Voxel coordinate system, hopefully it doesn't confuse you. And I will try your preprocessing code and do more experiment.

**Comment by zunzhumu** [comment]
> > 因为我在新公司中重建了CPMNet v2，所以这些培训代码侧重于特定的业务。LUNA2 的原始 CPMNet v16 代码与当前代码的样式不同，因此我没有将该代码拉入此存储库
> 
> Thank you for your answer. I noticed this code because of the article SCPM-net. What is the relationship between CPMV2 and SCPM? Is there any chance for the code of SCPMnet to be made public?

It wasn't meant to be public, but many people email to ask questions about the code. Compared to CPMNet v1, SCPMNet only added a sphere IoU loss. However, CPMNet v2 is a much stronger version of v1. For example, v2 only outputs predictions at a single scale rather than two scales like v1. If you are interested in this work, you can add this WeChat group to continue to communicate.
![image](https://github.com/zunzhumu/CPMNetv2/assets/24983216/80bc0f53-b2a6-4732-a5af-f7af5501e54c)

**Comment by WeebOwO** [comment]
Problem solved.

**Comment by sw1ngz** [comment]
Hi, now I want to reproduce the result in luna16.
Can I get a copy of your code?
My email is 1157801627@qq.com,appreciate it!

**Comment by sw1ngz** [comment]
Hi, now I want to reproduce the result in luna16.Can I get a copy of your code?
My email is 1157801627@qq.com,appreciate it!

**Comment by MartinCartajena** [comment]
> [@vincentwan0930](https://github.com/vincentwan0930) [@zxy630](https://github.com/zxy630) [@Pang-b0](https://github.com/Pang-b0) Hi, guys, I open my code at https://github.com/WeebOwO/LuCPMV2. I will try do some document work so that you can easy to get started. PS: There should be some bugs, and I hope you can help maintain the repo together😊


Hello, 

if you managed to recreate the training code for Luna, could I access it or could you share it with me?

Thank you very much!
